File: src/main/java/classloaders/PersonService.java
package classloaders;

/**
 * @author Evgeny Borisov
 */
public class PersonService {
   public static PersonService personService;

    static {
        System.out.println("Class was loaded");
        personService = new PersonService();
    }
    private PersonService() {
        System.out.println("Person was created");
    }
    public void doX(){
        System.out.println("XXXXXXXXX");
    }
}

File: src/main/scala/solutions/lab0/PersonService.scala
package solutions.lab0

import solutions.lab0.Main.Person

/**
  * @author Evgeny Borisov
  */
object PersonService {
  println("I was created")
  def processPerson(person:Person):Unit={
    println(person)
  }
  def dox()={}

  def apply(): Person = new Person(name = "Vasya")

  def main(args: Array[String]): Unit = {

    val john = new Person(age = 12,name = "John")
    val jack = new Person("Jack")
    val ivan = new Person(age = 30)
    val mali = new Person(name = "Mali")
  }
}

